src/aiida/engine/processes/workchains/restart.py

```diff
@@ -29,7 +29,7 @@
 
 
 def validate_handler_overrides(
-    process_class: 'BaseRestartWorkChain', handler_overrides: Optional[orm.Dict], ctx: 'PortNamespace'
+    process_class: type['BaseRestartWorkChain'], handler_overrides: Optional[orm.Dict], ctx: 'PortNamespace'
```

Comment:
Interesting. It might be worth adding a comment that explains why this is needed. What was the mypy error?

Reply:
It is

```
src/aiida/engine/processes/workchains/restart.py:160: error: Argument 1 to "validate_handler_overrides" has incompatible type "type[BaseRestartWorkChain]"; expected "BaseRestartWorkChain"  [arg-type]
```

Reply:
Ah, I see, now that I think about it more it makes sense, it's a type of that class, not a class instance, that is being passed down.